### ISSUE TITLE
about model architecture design

### ISSUE THREAD

**Issue by JunzheJosephZhu** [facebookresearch/dinov2#344]
two questions:
1. I noticed in vitl14.yaml, you set: 
```
dino:
  head_n_prototypes: 131072
  head_bottleneck_dim: 384
```
If I understand correctly, this is just a linear layer. Whats the reasoning behind the extreme ratio of input/output channels? Is it operating under the assumption that prototypes are trying to approximate a one-hot distribution, so it's 131072 bits vs 384 floating point numbers?

2. Do you have plans on releasing the prototype heads? I'm trying to adapt the model to food domain by continuing from your released checkpoints and further doing SSL on [recipe1M+](http://pic2recipe.csail.mit.edu/) which has 14M images. Ideally I could directly resume from the teacher/student model. If I understand correctly, currently released weights are only for teacher backbones, right?

**Comment by buaa-luzhi** [comment]
@JunzheJosephZhu 
Hello, I have the same idea as you. But I'm not sure whether the publicly available pre-training model is for the teacher network or the student network. Can you give me some advice
Thank you!

**Comment by buaa-luzhi** [comment]
Are there pre-trained weights for teacher or student networks?

**Comment by JunzheJosephZhu** [comment]
> @JunzheJosephZhu Hello, I have the same idea as you. But I'm not sure whether the publicly available pre-training model is for the teacher network or the student network. Can you give me some advice Thank you!

that is for the teacher network. I just initialized the student network using pretrained teacher network weight

**Comment by JunzheJosephZhu** [comment]
if cfg.MODEL.PRETRAINED_BACKBONE:
        state_dict = torch.load(cfg.MODEL.PRETRAINED_BACKBONE)
        source_num_patches = state_dict["pos_embed"].size(1) - 1
        source_h = source_w = int(source_num_patches ** 0.5)
        target_num_patches = model.student["backbone"].patch_embed.num_patches
        target_h = target_w = int(target_num_patches ** 0.5)
        dim = state_dict["pos_embed"].shape[-1]
        class_pos_embed = state_dict["pos_embed"][:, :1]
        patch_pos_embed = F.interpolate(
            state_dict["pos_embed"][:, 1:].reshape(1, source_h, source_w, -1).permute(0, 3, 1, 2),
            size=(target_h, target_w),
            mode="bicubic",
            antialias=model.student['backbone'].interpolate_antialias,
        )
        patch_pos_embed = patch_pos_embed.permute(0, 2, 3, 1).reshape(1, -1, dim)
        state_dict["pos_embed"] = torch.cat((class_pos_embed, patch_pos_embed), dim=1)
        
        # interpolate pos embed
        model.student['backbone'].load_state_dict(state_dict)
        model.teacher['backbone'].load_state_dict(state_dict)

**Comment by buaa-luzhi** [comment]
@JunzheJosephZhu ，  Thank you for your reply.
   if cfg.student.pretrained_weights:
            chkpt = torch.load(cfg.student.pretrained_weights)
            logger.info(f"OPTIONS -- pretrained weights: loading from {cfg.student.pretrained_weights}")
            student_backbone.load_state_dict(chkpt, strict=False)

In the SSLMetaArch method, is it possible to load weights for the student network in this way?

**Comment by JunzheJosephZhu** [comment]
yes, but make sure you also load for the teacher. Plus, the position embedding they provided in pretrained weights are for 512x512 images

**Comment by buaa-luzhi** [comment]
Yes, I have modified patch_size and global_crops_size to match the pre-trained model.
1、Furthermore, when training a model on a domain-specific dataset, which does not update the weights, the student network or the teacher network? As follows:
**"# there is no backpropagation through the teacher, so no need for gradients
for p in self.teacher.parameters():
     p.requires_grad = False
"**
2、Can the trained model be used to extract image features as follows?

transform = T.Compose([
    T.GaussianBlur(9, sigma=(0.1, 2.0)),
    T.Resize((patch_h * 14, patch_w * 14)),
    T.CenterCrop((patch_h * 14, patch_w * 14)),
    T.ToTensor(),
    T.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)),
])
 
dinov2_vitb14 = torch.hub.load('', 'dinov2_vits14',source='local').cuda()
print(dinov2_vitb14)

3、In /facebookresearch/dinov2， there is no public ‘’linear head‘’ training code， right？

Thank you again for your peer-to-peer response。

**Comment by JunzheJosephZhu** [comment]
1. The teacher model's weights is just EMA of student weights. 2. i dont think source="local" will work, that seems like just your local cache dir of the default pretrained weights. You have to manually do dinov2_vitb14.load("/path/to/custom_weights"). Also, I wouldn't put GaussianBlur and CenterCrop during inference, those are training augmentations 3. There is, it's under dinov2/eval

**Comment by buaa-luzhi** [comment]
1、For the first, how should the student network and teacher network weights change after loading the pre-trained model during the training of the custom dataset?Is it updated or not updated?
2、For the second,  you are right.
3、I found the 'linear head' code, and what I want to know is can I train the backbone with a 'linear head' layer behind the network while training the custom dataset?
